components/movementsensor/dualgps/dualgps.go

```diff
@@ -0,0 +1,236 @@
+) (movementsensor.MovementSensor, error) {
+	dg := dualGPS{
+		Named:  conf.ResourceName().AsNamed(),
+		logger: logger,
+	}
+
+	if err := dg.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	return &dg, nil
+}
+
+func (dg *dualGPS) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	dg.mu.Lock()
+	defer dg.mu.Unlock()
+
+	first, err := movementsensor.FromDependencies(deps, newConf.Gps1)
+	if err != nil {
+		return err
+	}
+	dg.gps1.gps = first
+
+	second, err := movementsensor.FromDependencies(deps, newConf.Gps2)
+	if err != nil {
+		return err
+	}
+	dg.gps2.gps = second
+
+	dg.offset = defaultOffsetDegrees
+	if newConf.Offset != 0 {
+		dg.offset = newConf.Offset
+	}
+
+	return nil
+}
+
+type gpsWithLock struct {
+	gps     movementsensor.MovementSensor
+	gpsLock sync.Mutex
+}
+
+func (lgps *gpsWithLock) getPosition(ctx context.Context, extra map[string]interface{}) (*geo.Point, error) {
+	lgps.gpsLock.Lock()
+	defer lgps.gpsLock.Unlock()
+	pos, _, err := lgps.gps.Position(ctx, extra)
+	return pos, err
+}
+
+// Position gets the position of a fake movementsensor.
+func (dg *dualGPS) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+	return geo.NewPoint(math.NaN(), math.NaN()), math.NaN(), movementsensor.ErrMethodUnimplementedPosition
+}
+
+// getHeading calculates bearing and absolute heading angles given 2 geoPoint coordinates
+// 0 degrees is North, 90 degrees is East, 180 degrees is South, 270 is West.
+func getHeading(first, second *geo.Point, yawOffset float64) (float64, float64, float64) {
+	// convert latitude and longitude readings from degrees to radians
+	// so we can use go's periodic math functions
+	firstLat := utils.DegToRad(first.Lat())
+	firstLong := utils.DegToRad(first.Lng())
+	secondLat := utils.DegToRad(second.Lat())
+	secondLong := utils.DegToRad(second.Lng())
+
+	// calculate the bearing between gps1 and gps2
+	deltaLong := secondLong - firstLong
+	y := math.Sin(deltaLong) * math.Cos(secondLat)
+	x := math.Cos(firstLat)*math.Sin(secondLat) - math.Sin(firstLat)*math.Cos(secondLat)*math.Cos(deltaLong)
+	bearing := utils.RadToDeg(math.Atan2(y, x))
+
+	// maps the bearing to the range of 0-360 degrees
+	if bearing < 0 {
+		bearing += 360
+	}
+
+	// calculate heading from bearing, accounting for yaw offset between the two gps devices
+	// e.g if the MovementSensor antennas are mounted on the left and right sides of the robot,
+	// the yaw offset would be roughly 90 degrees
+	heading := bearing - yawOffset
+	// make heading positive again
+	if heading < 0 {
+		diff := math.Abs(heading)
+		heading = 360 - diff
+	}
+
+	// return the standard bearing in the -180 to 180 range, with North being 0, East being 90
+	// South being 180/-180 and West being -90
+	var standardBearing float64
+	if bearing > 180 {
+		standardBearing = -(360 - bearing)
```

Comment:
This also looks needlessly complicated. I think it's just
```go
standardBearing := bearing
if standardBearing > 180 {
    standardBearing -= 360
}
```
...and now I wonder if we calculated the standardBearing on line 149, and this paragraph is redundant with lines 151-154.